File: go/config/Network.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Network 对应 config/Network.py。
type Network struct {
	*session.ApiGroup
}

// NewNetwork 创建 Network API 分组。
func NewNetwork(s *session.Session) *Network {
	return &Network{ApiGroup: session.NewApiGroup(s)}
}

// NetMode 对应 network/net-mode.xml。
func (n *Network) NetMode() (map[string]interface{}, error) {
	return n.S.Get("network/net-mode.xml", nil, "config")
}

// Networkmode 对应 network/networkmode.xml。
func (n *Network) Networkmode() (map[string]interface{}, error) {
	return n.S.Get("network/networkmode.xml", nil, "config")
}

// Config 对应 network/config.xml。
func (n *Network) Config() (map[string]interface{}, error) {
	return n.S.Get("network/config.xml", nil, "config")
}

// NetworkbandNull 对应 network/networkband_null.xml。
func (n *Network) NetworkbandNull() (map[string]interface{}, error) {
	return n.S.Get("network/networkband_null.xml", nil, "config")
}

// SetOnly4g 对应 network/setOnly4g.xml。
func (n *Network) SetOnly4g() (map[string]interface{}, error) {
	return n.S.Get("network/setOnly4g.xml", nil, "config")
}

File: src/config/network.rs
//! Network config group (`config/Network.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Network config group.
pub struct NetworkConfig<'a> {
    conn: &'a Connection,
}

impl<'a> NetworkConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        NetworkConfig { conn }
    }

    /// `network/net-mode.xml`. Network mode.
    pub fn net_mode(&self) -> Result<Value> {
        get_value(self.conn, "config/network/net-mode.xml")
    }

    /// `network/networkmode.xml`. Network mode.
    pub fn networkmode(&self) -> Result<Value> {
        get_value(self.conn, "config/network/networkmode.xml")
    }

    /// `network/config.xml`. Network configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/network/config.xml")
    }

    /// `network/networkband_null.xml`. Network band (null).
    pub fn networkband_null(&self) -> Result<Value> {
        get_value(self.conn, "config/network/networkband_null.xml")
    }

    /// `network/setOnly4g.xml`. Set only 4G.
    pub fn set_only_4g(&self) -> Result<Value> {
        get_value(self.conn, "config/network/setOnly4g.xml")
    }
}
